src/Compilers/Core/Portable/PEWriter/PeWriter.cs

```diff
@@ -26,21 +26,70 @@ public PeWritingException(Exception inner)
 
     internal static class PeWriter
     {
-        public static bool WritePeToStream(
+        internal static bool WritePeToStream(
+             EmitContext context,
+             CommonMessageProvider messageProvider,
+             Func<Stream> getPeStream,
+             Func<Stream> getMetadataPeStreamOpt,
+             Func<Stream> getPortablePdbStreamOpt,
+             PdbWriter nativePdbWriterOpt,
+             string pdbPathOpt,
+             bool metadataOnly,
+             bool isDeterministic,
+             CancellationToken cancellationToken)
+        {
+            if (!WritePeToStream(
+                context,
+                messageProvider,
+                getPeStream,
+                getPortablePdbStreamOpt,
+                nativePdbWriterOpt,
+                pdbPathOpt,
+                metadataOnly,
+                isDeterministic,
+                cancellationToken))
+            {
+                return false;
+            }
+
+            // produce the secondary output (ref assembly) if needed
```

Comment:
When you call the new `Emit` overload with both `peStream` and `metadataPeStream`. Also, when you call the command-line with `/refout:..`.